src/backend/joanie/core/api/__init__.py

```diff
@@ -19,7 +21,9 @@ def exception_handler(exc, context):
             detail = exc.message
         elif hasattr(exc, "messages"):
             detail = exc.messages
-
         exc = DRFValidationError(detail=detail)
+    elif isinstance(exc, TransitionNotAllowed):
+        detail = str(exc)
+        exc = PermissionDenied(detail=detail)
```

Comment:
I was hesitating with a 400 as it is not an access right problem but the context that makes it impossible?

Reply:
We can raise a ParseError which returns a 400